sdk/cognitivelanguage/azure-ai-language-questionanswering-authoring/migration_guide_authoring.md

```diff
@@ -0,0 +1,215 @@
+
+* `begin_update_sources`
+* `begin_update_qnas`
+* `begin_export`
+* `begin_import_method`
+* `begin_deploy_project`
+
+Future versions may introduce typed results; avoid relying on payload shape.
+
+### Example (Import)
+
+```python
+from azure.ai.language.questionanswering.authoring import QuestionAnsweringAuthoringClient, models as qa_models
+from azure.core.credentials import AzureKeyCredential
+
+client = QuestionAnsweringAuthoringClient(endpoint, AzureKeyCredential(key))
+assets = qa_models.ImportJobOptions(
+    assets=qa_models.Assets(
+        qnas=[
+            qa_models.ImportQnaRecord(
+                id=1,
+                answer="Example",
+                source="https://contoso/faq",
+                questions=["Example?"],
+            )
+        ]
+    )
+)
+poller = client.begin_import_method(project_name="MyProject", body=assets, format="json")
+poller.result()  # None (just completion)
+```
+
+ 
+## 7. Authentication Guidance
+
+Preferred: Azure Active Directory with `DefaultAzureCredential`.
+
+```python
+from azure.identity import DefaultAzureCredential
+from azure.ai.language.questionanswering.authoring import QuestionAnsweringAuthoringClient
+
+endpoint = "https://<resource-name>.cognitiveservices.azure.com"
+credential = DefaultAzureCredential()
+client = QuestionAnsweringAuthoringClient(endpoint, credential)
+```
+
+Fallback (resource key):
+
+```python
+from azure.core.credentials import AzureKeyCredential
+client = QuestionAnsweringAuthoringClient(endpoint, AzureKeyCredential(key))
+```
+
+Ensure you use the custom subdomain endpoint for AAD (not a regional alias) so the challenge flow succeeds.
+
+## 8. Python & Dependencies
+
+* Python >= 3.9
+* Core dependencies: `azure-core`, `isodate`, (conditionally) `typing-extensions` (<3.11)
+
+Upgrade from Python 3.8 before installing.
+
+## 9. Environment Variables (Optional Convenience)
+
+| Variable | Purpose |
+|----------|---------|
+| `AZURE_QUESTIONANSWERING_ENDPOINT` | Resource endpoint |
+| `AZURE_QUESTIONANSWERING_KEY` | Key credential (if not using AAD) |
+| `AZURE_QUESTIONANSWERING_PROJECT` | Optional: default project name for samples |
+
+ 
+## 10. Sample & Test Layout
+
+* `samples/` and `async_samples/` now contain **authoring-only** scenarios.
+* Runtime question querying samples were removed from this package.
+* Live test fixtures trimmed to authoring necessities only.
+
+ 
+## 11. Backward / Rollback Strategy
+
+If you still need runtime querying before a dedicated runtime package is published, you can temporarily continue using the combined package version you had. Plan to migrate fully once the runtime split is available.
+
+ 
+## 12. Synonyms, Sources, and QnAs Model-Based Updates
+
+Use model classes instead of raw dictionaries for clarity and forward compatibility:
+
+```python
+from azure.ai.language.questionanswering.authoring import models as qa_models
```

Comment:
please change all references of this pattern. Just import the exact models needed